x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncTests.java

```diff
@@ -103,62 +113,70 @@ public void testCreateRoundingNullInterval() {
-        // 7 days period should return weekly rounding
-        assertEquals(toMillis("2023-02-13T00:00:00.00Z"), process(ts, createRounding(Period.ofDays(7))));
-        // 3 months period should return quarterly
-        assertEquals(toMillis("2023-01-01T00:00:00.00Z"), process(ts, createRounding(Period.ofMonths(3))));
-
-        assertEquals(toMillis("2023-02-17T10:00:00.00Z"), process(ts, createRounding(Duration.ofHours(1))));
-        assertEquals(toMillis("2023-02-17T10:25:00.00Z"), process(ts, createRounding(Duration.ofMinutes(1))));
-        assertEquals(toMillis("2023-02-17T10:25:33.00Z"), process(ts, createRounding(Duration.ofSeconds(1))));
-
-        assertEquals(toMillis("2023-02-17T09:00:00.00Z"), process(ts, createRounding(Duration.ofHours(3))));
-        assertEquals(toMillis("2023-02-17T10:15:00.00Z"), process(ts, createRounding(Duration.ofMinutes(15))));
-        assertEquals(toMillis("2023-02-17T10:25:30.00Z"), process(ts, createRounding(Duration.ofSeconds(30))));
-        assertEquals(toMillis("2023-02-17T10:25:30.00Z"), process(ts, createRounding(Duration.ofSeconds(30))));
-
         IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> process(ts, createRounding(Period.ofDays(-1))));
         assertThat(e.getMessage(), containsString("Zero or negative time interval is not supported"));
 
         e = expectThrows(IllegalArgumentException.class, () -> process(ts, createRounding(Duration.ofHours(-1))));
         assertThat(e.getMessage(), containsString("Zero or negative time interval is not supported"));
     }
 
-    private static long toMillis(String timestamp) {
-        return Instant.parse(timestamp).toEpochMilli();
+    private static List<TestCaseSupplier> suppliers() {
+        long ts = toMillis("2023-02-17T10:25:33.38Z");
+        return List.of(
+            ofDatePeriod(Period.ofDays(1), ts, "2023-02-17T00:00:00.00Z"),
+            ofDatePeriod(Period.ofMonths(1), ts, "2023-02-01T00:00:00.00Z"),
+            ofDatePeriod(Period.ofYears(1), ts, "2023-01-01T00:00:00.00Z"),
+            ofDatePeriod(Period.ofDays(10), ts, "2023-02-12T00:00:00.00Z"),
+            // 7 days period should return weekly rounding
+            ofDatePeriod(Period.ofDays(7), ts, "2023-02-13T00:00:00.00Z"),
+            // 3 months period should return quarterly
+            ofDatePeriod(Period.ofMonths(3), ts, "2023-01-01T00:00:00.00Z"),
+            ofDuration(Duration.ofHours(1), ts, "2023-02-17T10:00:00.00Z"),
+            ofDuration(Duration.ofMinutes(1), ts, "2023-02-17T10:25:00.00Z"),
+            ofDuration(Duration.ofSeconds(1), ts, "2023-02-17T10:25:33.00Z"),
+            ofDuration(Duration.ofHours(3), ts, "2023-02-17T09:00:00.00Z"),
+            ofDuration(Duration.ofMinutes(15), ts, "2023-02-17T10:15:00.00Z"),
+            ofDuration(Duration.ofSeconds(30), ts, "2023-02-17T10:25:30.00Z"),
+            ofDuration(Duration.ofSeconds(30), ts, "2023-02-17T10:25:30.00Z")
+        );
```

Comment:
It'd be nice to have a random example, even if we can't assert that much interesting about it.